__tests__/__utils__/handlers.ts

```diff
@@ -223,36 +237,70 @@ function createDatabaseLayerHandler<
 >(args: {
   matchType: MessageType
   matchPayloads?: Partial<Payload>[]
-  resolver: MessageResolver<MessageType, Payload>
+  resolver: ResponseResolver<Record<string, unknown>, { payload: Payload }>
+  options?: { once: boolean }
 }) {
-  const { matchType, matchPayloads, resolver } = args
+  const { matchType, matchPayloads, resolver, options } = args
 
-  const handler = rest.post(getDatabaseLayerUrl({ path: '/' }), resolver)
+  return http.post(
+    getDatabaseLayerUrl({ path: '/' }),
+    withTypeAndPayload(resolver, matchType, matchPayloads) as ResponseResolver<
+      HttpRequestResolverExtras<PathParams>,
+      { payload: Payload },
+      undefined
+    >,
+    options,
+  )
+}
 
-  // Only use this handler if message matches
-  handler.predicate = (req: MockedRequest<BodyType<MessageType, Payload>>) =>
-    req?.body?.type === matchType &&
-    (matchPayloads === undefined ||
-      matchPayloads.some((payload) =>
-        R.equals({ ...req.body.payload, ...payload }, req.body.payload),
-      ))
+function withTypeAndPayload<
+  MessageType extends string = string,
+  Payload = DefaultPayloadType,
+>(
+  resolver: ResponseResolver<Record<string, unknown>, { payload: Payload }>,
+  expectedType: MessageType,
+  expectedPayloads?: Partial<Payload>[],
+) {
```

Comment:
```suggestion
): ResponseResolver {
```

Then the type definitions like `HttpRequestResolverExtras<PathParams>` are not needed any more